packages/compiler-core/src/transforms/vModel.ts

```diff
@@ -1 +1,54 @@
-// TODO
+import { DirectiveTransform } from '../transform'
+import {
+  createSimpleExpression,
+  createObjectProperty,
+  createCompoundExpression,
+  NodeTypes,
+  Property
+} from '../ast'
+import { createCompilerError, ErrorCodes } from '../errors'
+import { isEmptyExpression } from '../utils'
+
+export const transformModel: DirectiveTransform = (dir, node, context) => {
+  const { exp, arg } = dir
+  if (!exp) {
+    context.onError(createCompilerError(ErrorCodes.X_V_MODEL_NO_EXPRESSION))
+
+    return createTransformProps()
+  } else if (isEmptyExpression(exp)) {
+    context.onError(
+      createCompilerError(ErrorCodes.X_V_MODEL_MALFORMED_EXPRESSION)
+    )
+
+    return createTransformProps()
+  } else {
+    const defaultPropName = createSimpleExpression('modelValue', true)
+    const defaultEventName = createSimpleExpression('onUpdate:modelValue', true)
+    const propName = arg ? arg : defaultPropName
```

Comment:
```suggestion
    const propName = arg ? arg : createSimpleExpression('modelValue', true)
```

Inlining the default value here avoids creating them when they are not needed. Same for `defaultEventName`.